File: Javascript/Semana10/Empleado.js
class Empleado extends Persona{

    static contadorEmpleados = 0;

    constructor(nombre, apellido, edad, sueldo){
        super(nombre, apllido, edad);
        this._idEmpleado = ++Empleado.contadorEmpleados;
        this._sueldo = sueldo;
    }

    get _idEmpleado(){
        return this._idEmpleado;
    }
    
    get sueldo(){
        this._sueldo;
    }

    set sueldo(sueldo){
        this._sueldo = sueldo;
    }
    toString(){
        return `
        ${super.toString}
        ${this._idEmpleado}
        ${this._sueldo}`;
    }
}

File: Java/Semana5/ConversionObjetos/src/domain/Empleado.java

package domain;

public class Empleado { //clase padre
    protected String nombre;
    protected double sueldo;

    public Empleado(String nombre, double sueldo){ //constructor
	this.nombre = nombre; //instancias de los atributos
	this.sueldo = sueldo; 
    } 

    // metodo para la sobreescritura
    public String obtenerDetalles(){
	return "Nombre: "+this.nombre+", Sueldo: "+this.sueldo;
    } 

    public String getNombre() {
	return nombre;
    }

    public void setNombre(String nombre) {
	this.nombre = nombre;
    }

    public double getSueldo() {
	return sueldo;
    }

    public void setSueldo(double sueldo) {
	this.sueldo = sueldo;
    } 

    @Override
    public String toString(){
	return "Empleado{"+ "nombre = "+ nombre + ", sueldo = "+ sueldo +"}";
}

    
}
